### DISCUSSION TITLE
Bagaimana cara mengekstrak nilai korelasi dari grafik korelasi ke dalam vektor?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#116] [Q&A] [answered]
Misalkan, saya memiliki grafik korelasi dari ggally sebagai berikut, 
apakah ada cara untuk mengekstrak nilai korelasi salah satu predictor sebagai vektor, beserta label nilai korelasi tersebut berpasangan dengan predictor apa? 
![Capture](https://user-images.githubusercontent.com/95233522/170445131-979374ca-9b00-4ed4-b106-8cdfaf3d7f0f.JPG)

**Comment by onlyphantom** [comment, marked as answer]
The following should return a named vector of correlation between `goals` and all variables, assuming dataframe is named `df`:

```
cor(df$goals, df)
```

**Reply by Hertzspr** [reply]
It says, y must be numeric.
![fail](https://user-images.githubusercontent.com/95233522/170658990-24738891-8b3e-4d0a-83db-a961010f7b24.JPG)

**Reply by Hertzspr** [reply]
Wait, I think your answer was actually right, it's just, that my data frame has non numeric column. 

So I tried:
```
i1 <- sapply(haaland_dortmund_scoring, is.numeric)

y1 <- "goals" #change it to actual column name

x1 <- setdiff(names(haaland_dortmund_scoring)[i1], y1)

cor(haaland_dortmund_scoring[x1], haaland_dortmund_scoring[[y1]])
```

and it's done! Thanks.
